### PR TITLE
Add component.yaml validator 

### PR DESCRIPTION
**Pull request by binoyPeries**
This Pr introduces a new Github Action to validate choreo source config file (component.yaml) that can be used in choreo build flow
Related: https://github.com/wso2-enterprise/choreo/issues/30039

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules/
```
Comment (yashodgayashan) [inline .gitignore]:
```suggestion
node_modules/

```

File: README.md [hunk 2]
```diff
@@ -1 +1,26 @@
...
+
+The result of the validation
+
+## Example usage
+
+```yaml
+build:
+  steps:
+    - name: Choreo Source Configuration Validator
+      uses: choreo-templates/choreo-source-config-validator@v1
```
Comment (yashodgayashan) [inline README.md]: Lets go with following convention for now

```suggestion
      uses: choreo-templates/choreo-source-config-validator@v0.1.0
```

File: README.md [hunk 3]
```diff
@@ -1 +1,26 @@
...
+      with:
+        source-root-dir: ${{source-root-dir}}
```
Comment (yashodgayashan) [inline README.md]:
```suggestion
        source-root-dir-path: ${{source-root-dir}}
```

File: action.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+name: "Choreo Source Configuration Validator"
+description: "Validate the source configuration files of Choreo"
+inputs:
+  source-root-dir: # the path to root directory of the source code
```
Comment (yashodgayashan) [inline action.yml]:
```suggestion
  source-root-dir: 
```

File: index.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+const core = require("@actions/core");
+const fs = require("fs");
+const yaml = require("js-yaml");
+const { componentYamlSchema } = require("./schemas");
+
+// util functions
```
Comment (yashodgayashan) [inline index.js]:
```suggestion
```

File: index.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+// Main code
+let sourceRootDir = "";
+try {
+  sourceRootDir = core.getInput("source-root-dir");
+} catch (error) {
+  console.log("Failed to read input:", error.message);
+  core.setFailed("Failed to read input", error.message);
+}
```
Comment (yashodgayashan) [inline index.js:32]:
```suggestion
}

```

**Review by yashodgayashan — CHANGES_REQUESTED** [review]

File: index.js (hunk 6 above)
Comment (binoyPeries) [reply index.js:32]: Left a space after the catch block

File: schemas.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+});
+// basePathRequired - Custom validation method to check base path is required for REST, GraphQL, and WS endpoints
+yup.addMethod(yup.object, "basePathRequired", function () {
+  return this.test({
+    name: "base-path-required",
+    test: (value, testCtx) => {
+      const { type } = testCtx.parent;
+      if (BASE_PATH_REQUIRED_TYPES.includes(type) && !value.basePath) {
+        return new yup.ValidationError(
+          "base path is required for REST, GraphQL, and WS endpoints"
```
Comment (Nashaath) [inline schemas.js]:
```suggestion
          "Base path is required for REST, GraphQL, and WS endpoints"
```

Comment (binoyPeries) [reply schemas.js]: Updated

**Review by Nashaath — APPROVED** [review]

**Review by yashodgayashan — APPROVED** [review]

**Pull request merged**